lib/HttpClient/CurlClient.php

```diff
@@ -147,6 +147,15 @@ public function request($method, $absUrl, $headers, $params, $hasFile)
             throw new Error\Api("Unrecognized method $method");
         }
 
+        // It is only safe to retry network failures on post and delete
+        // requests if we add an Idempotency-Key header
+        if ((($method == 'post') || ($method == 'delete')) &&
+            (Stripe::$maxNetworkRetries > 0)) {
+            if (!isset($headers['Idempotency-Key'])) {
+                $headers['Idempotency-Key'] = bin2hex(openssl_random_pseudo_bytes(32));
```

Comment:
Agreed. I've actually already updated `RandomGenerator` with a `uuid` method that returns a real V4 UUID since that's the best practice recommended in [our documentation](https://stripe.com/docs/api#idempotent_requests) (and what's done in stripe-ruby too).

Reply:
Nice! Love the UUID generator.